packages/db-base/lib/inMemFileDB.js

```diff
@@ -264,50 +271,81 @@ class InMemoryFileDB {
         return text;
     }
 
+    /**
+     * Handle saving the dataset incl. backups
+     */
     saveState() {
+        const jsonString = this.saveDataset();
+
+        if (!this.settings.backup.disabled && jsonString) {
+            this.saveBackup(jsonString);
+        }
+
+        if (this.stateTimer) {
+            clearTimeout(this.stateTimer);
+            this.stateTimer = null;
+        }
+    }
+
+    /**
+     * Saves the dataset into File incl. handling of a fallback backup file
+     *
+     * @returns {string} JSON string of the complete dataset to also be stored into a compressed backup file
+     */
+    saveDataset() {
         try {
             if (fs.existsSync(this.datasetName)) {
-                const old = fs.readFileSync(this.datasetName);
-                fs.writeFileSync(this.datasetName + '.bak', old);
+                fs.renameSync(this.datasetName, `${this.datasetName}.bak`);
```

Comment:
We should use the overwrite option. I've learned that the hard way (.bak exists -> crash)

Reply:
Kk